operator/v1/types_console.go

```diff
@@ -202,6 +205,49 @@ type AddPage struct {
 	DisabledActions []string `json:"disabledActions,omitempty"`
 }
 
+type PerspectiveState string
+
+// These are valid perspective states. "Enabled" means the perspective is shown.
+// "Disabled" means the Perspective is hidden.
+// "AccessReview" means access review check is required to show or hide a Perspective.
+const (
+	Enabled      PerspectiveState = "Enabled"
+	Disabled     PerspectiveState = "Disabled"
+	AccessReview PerspectiveState = "AccessReview"
+)
+
+type PerspectiveAccessReview struct {
+	// requiresAccessReview defines a list of permission checks. The perspective will only be shown when all checks are successful.
+	// +optional
+	RequiresAccessReview []authorizationv1.ResourceAttributes `json:"requiresAccessReview,omitempty"`
+	// missingAccessReview defines a list of permission checks. The perspective will only be shown when at least one check fails.
+	// +optional
+	MissingAccessReview []authorizationv1.ResourceAttributes `json:"missingAccessReview,omitempty"`
+}
+
+// PerspectiveVisibility defines the criteria to show/hide a perspective
+// +union
+type PerspectiveVisibility struct {
+	// state defines the perspective is enabled or disabled or access review check is required.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Enabled;Disabled;AccessReview
+	// +kubebuilder:validation:Required
+	State PerspectiveState `json:"state,omitempty"`
+	// accessReview defines required and missing access review checks.
+	// +optional
+	AccessReview PerspectiveAccessReview `json:"accessReview,omitempty"`
+}
+
+type Perspectives struct {
+	// id defines the id of perspective to disable.
+	// Incorrect or unknown ids will be ignored.
+	// +kubebuilder:validation:Required
+	ID string `json:"id"`
```

Comment:
Is there any validation we can add here? Is there a set of characters or a length limit we can impose?
Where can a user find these values?

Reply:
currently, we do not have any criteria for id in our javascript code. The id here should match the extension id of the perspective. So leaving this as is.

Reply:
Where does a user find out the ID? We should make a note in the documentation to give them a hint where they might find this